src/App.js

```diff
@@ -14,11 +18,126 @@ const goodsFromServer = [
   'Garlic',
 ];
 
-const App = () => (
-  <div className="App">
-    <h1>Goods</h1>
-    {goodsFromServer.length}
-  </div>
-);
+class App extends React.Component {
+  state = {
+    goods: [],
+    visibleGoods: [],
+    value: '1',
+    displayClass: false,
+  }
+
+  showGoods = () => {
+    this.setState({
+      goods: goodsFromServer,
+      visibleGoods: goodsFromServer,
+      displayClass: true,
+    });
+  }
+
+  reverse = () => {
+    this.setState(state => ({ visibleGoods: [...state.visibleGoods].reverse() }));
+  }
+
+  sortByAlpha = () => {
+    this.setState(state => ({
+      visibleGoods: [...state.visibleGoods].sort((productA, productB) => productA.localeCompare(productB)),
+    }));
+  }
+
+  sortByLength = () => {
+    this.setState(state => ({
+      visibleGoods: [...state.visibleGoods].sort((productA, productB) => productA.length - productB.length),
+    }));
+  }
+
+  reset = () => {
+    this.setState(state => ({
+      visibleGoods: [...state.goods], value: '1',
+    }));
+  }
+
+  handleChange = (event) => {
+    this.setState({ value: event.target.value });
+  }
+
+  handleSubmit = (event) => {
+    alert(`Length of the good is set to: ${this.state.value} or more letters length`);
+    event.preventDefault();
+    this.setState(state => ({
+      visibleGoods: state.goods.filter(product => product.length >= parseInt(state.value, 10)),
+    }));
+  }
+
+  render() {
+    const { visibleGoods, displayClass } = this.state;
+
+    return (
+      <div className="App">
+        <h1>
+          Goods List Length:
+          {' '}
+          {' '}
+          {visibleGoods.length}
+        </h1>
+        <List goods={visibleGoods} />
+        <div className="Container">
+          <button
+            className={classNames({ Action: displayClass === true })}
```

Comment:
```suggestion
            className={cn({ Action: displayClass === true })}
```
його зазвичай ось так скорочують